apps/website/src/components/forms/GiveawayChecks.tsx

```diff
@@ -15,8 +16,46 @@ const GiveawayCheck: React.FC<{
   url?: string;
 }> = ({ name, label, children, url }) => {
   const [isClicked, setIsClicked] = useState(false);
+  const hasLink = url !== undefined;
 
-  const needsToBeClicked = url !== undefined;
+  // If this is a link, and the user has already clicked once,
+  // then just the children should be the link (not the checkbox)
+  const childrenJsx =
+    hasLink && isClicked ? (
+      <Link
+        rel="noreferrer"
+        target="_blank"
+        href={url}
+        className="flex flex-1 flex-row items-center gap-5 p-5 underline"
+      >
+        {children}
+      </Link>
+    ) : (
+      <span className="flex flex-1 flex-row items-center gap-5 p-5">
+        {children}
+      </span>
+    );
+
+  // Disable the checkbox if this is a link, and the user hasn't clicked yet
+  const disabled = hasLink && !isClicked;
+  const checkboxJsx = (
+    <label
+      className={classes(
+        "flex cursor-pointer flex-row items-center px-5",
+        disabled ? "pointer-events-none bg-gray-200" : "bg-white"
+      )}
+    >
+      <span className="sr-only">{label}</span>
+      <input
+        name={`req-${name}`}
+        value="yes"
+        type="checkbox"
+        defaultChecked={hasLink && isClicked}
+        required={true}
+        disabled={disabled}
```

Comment:
The problem with disabling them is that the browser form validation ignores the checkboxes here. Thats why i did the weird `preventDefault()` thing.

Reply:
Ah that makes sense, didn't notice it wasn't validating it when disabled